### DISCUSSION TITLE
Map verification?

### DISCUSSION THREAD

**Discussion by SuperiorBun** [ppy/osu#18279] [Ideas]
Dear peppy
So when you edit ( You can currently only delete notes ) a beatmap and delete notes so you can practice that one specific part, there's no way to redownload or make the beatmap the official one again, tried deleting and redownloading, seems like it kept the data somewhere on the PC and doesn't have the option to reverify a map.

**Comment by bdach** [comment]
For now, delete the map and restart the game. After that you should be able to redownload the original map again.

This will be better once issues like https://github.com/ppy/osu/issues/11922 and https://github.com/ppy/osu/issues/17676 are fixed, I believe.

**Reply by peppy** [reply]
Very weird. A restart should not be required, but something seems to be a bit odd. Can reproduce this, so will take a look into whether it can be fixed:

```
2022-05-15 14:52:22 [verbose]: [df7e5] Found existing (optimised) beatmap for unknown artist - unknown title (ID c6dfb462-d3d5-428b-baa0-38d58ae3a3ca) – skipping import.
2022-05-15 14:52:22 [verbose]: [df7e5] Existing beatmap's deletion flag has been removed to allow for reuse.
```

**Reply by bdach** [reply]
It's likely because online ID isn't being reset on save.

I'm thinking this path is being hit:

https://github.com/ppy/osu/blob/328561f5c816088b90fb4454131be031f2d2d83a/osu.Game/Stores/BeatmapImporter.cs#L146-L152

Or this one:

https://github.com/ppy/osu/blob/328561f5c816088b90fb4454131be031f2d2d83a/osu.Game/Stores/BeatmapImporter.cs#L154-L164
